highfidelity/hifi
# Cleanup Web3DOverlay and tablet home button

- Remove extraneous "resolution" field from web overlays.  This fixes a culling issue and makes them more closely match web entities.  Before, web overlays would render with the wrong dimensions.
- Copy intersection logic from web entities to web overlays to get rid of a TODO.
- Update tablet scripts to account for overlay changes.
- Fix positioning of tablet screen, home button, and highlight at different avatar scales ([FB9035](https://highfidelity.fogbugz.com/f/cases/9035) and probably [FB8906](https://highfidelity.fogbugz.com/f/cases/8906)).
- Replace the home button overlay with a circle overlay ([FB8214](https://highfidelity.fogbugz.com/f/cases/8214)).
- Fix reticle depth at different avatar scales.
- Fix tablet "x" button triggering on mouse click.

Test plan:
- Run [this](https://gist.githubusercontent.com/SamGondelman/aef57ea4aa6304a3149aab964d4865dc/raw/7b50ff9333f65a9f27e5bb9070b4e7610bbe3a02/WebOverlayTest.js).  It'll make a web overlay and a web entity in front of you.  Verify that they are the same size.
- Press '1' and '2' to increase and decrease their "dpi" settings.  Their dimensions should not change.
- Enable tablet in desktop mode (Developer -> UI).  You should be able to mouse over the tablet and the home button and have them highlight as expected.
- Change avatar scales.  The tablet screen and highlight effect should resize properly and not be offset from the expected positions.  Try a bunch of different scales.
- In HMD, lasers, mouse, and stylus should all work on the tablet and home button.  The highlight should be in the right place.  Try at different avatar scales.
- Everything (interacting with tablet, hovering over home button, etc) should still work when you grab the tablet with one hand and use the other hand to interact with it.
- If you look at the tablet from the side and laser onto the home button, the laser should be hitting very close to the surface of the tablet.  There should not be a noticeable space in between.
- The mouse depth and size should be correct at different avatar scales.
- With the mouse, the "x" button should only trigger on mouse release, not mouse press.
